cockatrice/src/client/network/replay_timeline_widget.cpp

```diff
@@ -81,16 +86,53 @@ void ReplayTimelineWidget::skipToTime(int newTime)
     }
 
     newTime -= newTime % 200; // Time should always be a multiple of 200
-    if (newTime < currentTime) {
-        currentTime = 0;
-        currentEvent = 0;
-        emit rewound();
+
+    const bool isBackwardsSkip = newTime < currentTime;
+    currentTime = newTime;
+
+    if (isBackwardsSkip) {
+        handleBackwardsSkip(doRewindBuffering);
+    } else {
+        processNewEvents();
     }
-    currentTime = newTime - 200; // 200 is added back in replayTimerTimeout
-    replayTimerTimeout();
+
     update();
 }
 
+/// @param doRewindBuffering When true, if multiple backward skips are made in quick succession, only a single rewind
+/// is processed at the end. When false, the backwards skip will always cause an immediate rewind
+void ReplayTimelineWidget::handleBackwardsSkip(bool doRewindBuffering)
+{
+    if (doRewindBuffering) {
+        // We use a one-shot timer to implement the rewind buffering.
+        // The rewind only happens once the timer runs out.
+        // If another backwards skip happens, the timer will just get reset instead of rewinding.
+        rewindBufferingTimer->stop();
+        rewindBufferingTimer->start(calcRewindBufferingTimeout());
+    } else {
+        // otherwise, process the rewind immediately
+        processRewind();
+    }
+}
+
+/// The timeout scales based on the current event number, up to a limit
+int ReplayTimelineWidget::calcRewindBufferingTimeout() const
+{
+    int extraTime = std::min(currentEvent / 100, 100);
+    return BASE_REWIND_BUFFERING_TIMEOUT_MS + extraTime;
```

Comment:
why have both a base and a minimum added to each other? why is the minimum not a constant?

Reply:
My intention is to have the total rewind timeout start at the base timeout and then scale up based on the number of events, up to a maximum of +100. Is there a better way to write that? 

I could make 100 into a `MAX_EXTRA_TIMEOUT` constant i guess.